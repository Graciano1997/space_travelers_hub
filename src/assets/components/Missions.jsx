import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/MissionsSlice';
import Mission from './Mission';
import styles from '../style/missions.module.css';
// import style from '../style/Info.module.css';

function Missions() {
  const dispatch = useDispatch();
  const { missions, error } = useSelector((state) => state.missions);

  // const error = useSelector((state) => state.missions.error);
  // const missions = useSelector((state) => state.missions.missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  // const joinHandler = (mission) => {
  //   // dispatch(joinMission(mission));
  //   dispatch(joinMission(mission));
  // };

  // const leaveHandler = (mission) => {
  //   dispatch(leaveMission(mission));
  // };

  // const renderMissions = () => {
  if (error) {
    return <div className={styles.error}>{error}</div>;
  }

  // if (loading) {
  //   return (
  //     <div className={style.isLoading}>
  //       <h2>Loading👨‍🚀👩🏿‍🚀...</h2>
  //     </div>
  //   );
  // }

  const missionsAvailable = missions.slice(0, 5);

  return (
    <table className={styles['mission-table']}>
      <thead>
        <tr>
          <th className={styles['mission-name']}>Mission</th>
          <th className={styles['mission-description']}>Description</th>
          <th className={styles['mission-status']}>Status</th>
          <th className={styles['mission-button']}>Actions</th>
        </tr>
      </thead>
      <tbody>
        {missionsAvailable.map((mission) => (
          <Mission
            mission={mission}
            // join={joinHandler(mission.mission_id)}
            // leave={leaveHandler(mission.mission_id)}
            key={mission.mission_id}
              // mobile={false}
            styles={styles}
          />
        ))}
      </tbody>
    </table>
  );
}

// return (
//   <div className={`w-100 p-4 ${styles.header}`}>
//     <div className={styles.missions}>
//       <h2>Missions</h2>
//     </div>
//     {renderMissions()}
//   </div>
// );
// }

export default Missions;
