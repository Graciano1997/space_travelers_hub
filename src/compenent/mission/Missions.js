import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions, joinMission, leaveMission } from '../../assets/redux/missions/MissionsSlice';
import Mission from './Mission';
import styles from '.../assets/style/missions.module.css';

function Missions() {
  const dispatch = useDispatch();
  const error = useSelector((state) => state.missions.error);
  const missions = useSelector((state) => state.missions.missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  const joinHandler = (mission) => {
    dispatch(joinMission(mission));
  };

  const leaveHandler = (mission) => {
    dispatch(leaveMission(mission));
  };

  const renderMissions = () => {
    if (error) {
      return <div className={styles.error}>{error}</div>;
    }

    const missionsAvailable = missions.slice(0, 8);

    return (
      <table className={styles['mission-table']}>
        <thead>
          <tr>
            <th className={styles['mission-name']}>Mission</th>
            <th className={styles['mission-description']}>Description</th>
            <th className={styles['mission-status']}>Status</th>
            <th className={styles['mission-button']}>Actions</th>
          </tr>
        </thead>
        <tbody>
          {missionsAvailable.map((mission) => (
            <Mission
              mission={mission}
              join={joinHandler}
              leave={leaveHandler}
              key={mission.mission_id}
              mobile={false}
              styles={styles}
            />
          ))}
        </tbody>
      </table>
    );
  };

  return (
    <div className={`w-100 p-4 ${styles.header}`}>
      <div className={styles.missions}>
        <h2>Missions</h2>
      </div>
      {renderMissions()}
    </div>
  );
}

export default Missions;
