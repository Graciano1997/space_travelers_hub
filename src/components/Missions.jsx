import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { fetchMissions } from '../redux/missions/MissionsSlice';
import Mission from './Mission';
import styles from '../assets/style/missions.module.css';
import style from '../assets/style/Info.module.css';

function Missions() {
  const dispatch = useDispatch();
  const { missions, error, loading } = useSelector((state) => state.missions);

  useEffect(() => {
    dispatch(fetchMissions());
  }, [dispatch]);

  if (error) {
    return <div className={styles.error}>{error}</div>;
  }

  if (loading) {
    return (
      <div className={style.isLoading}>
        <h2>Loading👨‍🚀👩🏿‍🚀...</h2>
      </div>
    );
  }

  const missionsAvailable = missions;

  return (
    <table className={styles['mission-table']}>
      <thead>
        <tr>
          <th className={styles['mission-name']}>Mission</th>
          <th className={styles['mission-description']}>Description</th>
          <th className={styles['mission-status']}>Status</th>
          <th className={styles['mission-button']}>{' '}</th>
        </tr>
      </thead>
      <tbody>
        {missionsAvailable.map((mission) => (
          <Mission
            mission={mission}
            key={mission.mission_id}
            styles={styles}
          />
        ))}
      </tbody>
    </table>
  );
}

export default Missions;
