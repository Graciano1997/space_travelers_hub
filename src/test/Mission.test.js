import { Provider } from 'react-redux';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import store from '../redux/store';
import configureMockStore from 'redux-mock-store';
import Missions from '../components/Missions';

// const mockStore = configureMockStore([]);

jest.mock('axios', () => ({
  get: jest.fn(),
}));

// describe('Loading missions component', () => {
//   it('renders correctly', () => {

//     const { tree } = render(
//       <Provider store={store}>
//         <Missions />
//       </Provider>,
//     );
//     expect(tree).toMatchSnapshot();
//   });
// });
// const initialState = {
//   missions: [
//     {
//       id: 1,
//       name: 'Mission',
//       description: 'This is a lengthy description indicating the mission paramaters',
//       reserved: false,
//     },
//   ],
// };


function MockingMissions() {
  // const store = mockStore(initialState);
  return (
    <Provider store={store}>
      <Missions />
    </Provider>
  );
}

test('the amount of  Rockets and implicitly the getRockets() ', () => {

  render(<MockingMissions />);
  const MissionElements = screen.getByText(/This is a lengthy /i);
  expect(MissionElements).toBeInTheDocument();
});

//  test('When the Rocket is reserved', async () => {
//    render(<MockingRockets />);
//    const reservedRockeButton = await screen.findByRole('button');
//    fireEvent.click(reservedRockeButton);
//    const cancelRockeButton = screen.getByText(/Cancel Reservation/i);
//    expect(cancelRockeButton).toBeInTheDocument();
//  });

//  test('When the Rockets Reserve is cancelled', async () => {
//    render(<MockingRockets />);
//    const reservedRockeButton = await screen.findByRole('button');
//    fireEvent.click(reservedRockeButton);
//    const cancelRockeButton = screen.getByText(/Cancel Reservation/i);
//    expect(cancelRockeButton).toBeInTheDocument();
//    fireEvent.click(reservedRockeButton);
//    const reserveRockeButton = screen.getByText(/Reserve Rocket/i);
//    expect(reserveRockeButton).toBeInTheDocument();
//  });

