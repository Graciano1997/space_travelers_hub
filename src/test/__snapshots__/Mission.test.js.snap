// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check if Evething is rendered correctly 1`] = `undefined`;

exports[`renders correctly 1`] = `undefined`;

exports[`testing the Missions Check if Evething is rendered correctly 1`] = `undefined`;
